src/plugins/legacy_plugin.cpp

```diff
@@ -47,55 +48,43 @@ void LegacyPlugin::ProcessQueryFunction(string function, char* buffer)
 
 void LegacyPlugin::GetFunctionClass(char* fClass)
 {
-	fClass = NULL;
+	assert(fClass);
+	fClass[0] = '\0';
 }
 
-char* LegacyPlugin::GetPluginFileName()
+const char* LegacyPlugin::GetPluginFileName()
 {
-	return pluginFileName;
+	return pluginFileName.c_str();
 }
 
-char* LegacyPlugin::GetPluginFullPath()
+const char* LegacyPlugin::GetPluginFullPath()
 {
-	return pluginFullPath;
+	return pluginFullPath.c_str();
 }
 
-void LegacyPlugin::SetPluginFullPath(char* fileName)
+void LegacyPlugin::SetPluginFullPath(std::string_view p)
 {
-	// TODO: replace with _splitpath
-	pluginFullPath = strdup(fileName);
+	const auto fullPath = std::filesystem::path(p);
+	pluginFullPath = fullPath.string();
 
-	// extract filename from full path
-	int len = (int)strlen(fileName) - 1;
-	if (len > 0)
+	if (fullPath.has_stem())
 	{
-		int begin = -1, end = -1;
-		for (int i = len; i >= 0; i--)
-		{
-			if ((end == -1) && (fileName[i] == '.'))
-				end = i;
-			else if ((begin == -1) && (fileName[i] == '\\'))
-				begin = i + 1;
-		}
-
-		if (end > begin)
-		{
-			pluginFileName = new char[MAX_PATH];
-			strncpy(pluginFileName, fileName + begin, min(MAX_PATH,  end - begin + 1));
-		}
+		pluginFileName = fullPath.stem().string();
 	}
 }
 
-void LegacyPlugin::nwnxcpy(char* buffer, const char* response)
+void LegacyPlugin::nwnxcpy(char* buffer, std::string_view response)
 {
-	nwnxcpy(buffer, response, strlen(response));
+	assert(buffer);
```

Comment:
Previous implementation was copying **up to** MAX_BUFFER characters (including \0 terminator), however in your implementation, if I'm not mistaken, it always copy MAX_BUFFER characters, even if the string is shorter.

`strncpy(buffer, response, MAX_BUFFER);` would copy the string and stop at the first `\0`, while also preventing going past MAX_BUFFER.

Same remark for the plugin.cpp implementation